internal/webhook/serving/v1alpha1/inferencegraph_webhook.go

```diff
@@ -71,3 +78,103 @@ func (d *InferenceGraphCustomDefaulter) Default(ctx context.Context, obj runtime
 
 	return nil
 }
+
+// NOTE: The 'path' attribute must follow a specific pattern and should not be modified directly here.
+// Modifying the path for an invalid path can cause API server errors; failing to locate the webhook.
+// +kubebuilder:webhook:path=/validate-serving-kserve-io-v1alpha1-inferencegraph,mutating=false,failurePolicy=fail,sideEffects=None,groups=serving.kserve.io,resources=inferencegraphs,verbs=create;update,versions=v1alpha1,name=vinferencegraph-v1alpha1.odh-model-controller.opendatahub.io,admissionReviewVersions=v1
+
+// InferenceGraphCustomValidator struct is responsible for validating the InferenceGraph resource
+// when it is created, updated, or deleted.
+//
+// NOTE: The +kubebuilder:object:generate=false marker prevents controller-gen from generating DeepCopy methods,
+// as this struct is used only for temporary operations and does not need to be deeply copied.
+type InferenceGraphCustomValidator struct{}
+
+var _ webhook.CustomValidator = &InferenceGraphCustomValidator{}
+
+// ValidateCreate implements webhook.CustomValidator so a webhook will be registered for the type InferenceGraph.
+func (v *InferenceGraphCustomValidator) ValidateCreate(_ context.Context, obj runtime.Object) (admission.Warnings, error) {
+	inferencegraph, ok := obj.(*servingv1alpha1.InferenceGraph)
+	if !ok {
+		return nil, fmt.Errorf("expected a InferenceGraph object but got %T", obj)
+	}
+	inferencegraphlog.Info("Validation for InferenceGraph upon creation", "name", inferencegraph.GetName())
+	return validateInferenceGraph(inferencegraph)
+}
+
+// ValidateUpdate implements webhook.CustomValidator so a webhook will be registered for the type InferenceGraph.
+func (v *InferenceGraphCustomValidator) ValidateUpdate(_ context.Context, _, newObj runtime.Object) (admission.Warnings, error) {
+	inferencegraph, ok := newObj.(*servingv1alpha1.InferenceGraph)
+	if !ok {
+		return nil, fmt.Errorf("expected a InferenceGraph object for the newObj but got %T", newObj)
+	}
+	inferencegraphlog.Info("Validation for InferenceGraph upon update", "name", inferencegraph.GetName())
+	return validateInferenceGraph(inferencegraph)
+}
+
+// ValidateDelete implements webhook.CustomValidator so a webhook will be registered for the type InferenceGraph.
+func (v *InferenceGraphCustomValidator) ValidateDelete(_ context.Context, _ runtime.Object) (admission.Warnings, error) {
+	// ODH does not do any validations when an InferenceGraph is being deleted
+	return nil, nil
+}
+
+func validateInferenceGraph(inferenceGraph *servingv1alpha1.InferenceGraph) (admission.Warnings, error) {
+	clusterDomainName := network.GetClusterDomainName()
```

Comment:
I assume that for now IG will be supported only with the Serverless mode, right?
Or does this function returns the cluster domain name for non knative resource?

Reply:
The plan is to also support Raw.
Although this func is coming from Knative libraries, I think its code is not generic enough for using here.

Check the small code here: https://github.com/knative/pkg/blob/3386f371effa5c5d8ab959895cc4ad0c6703e1a6/network/domain.go#L46